tests/http1_tests.py

```diff
@@ -264,46 +264,52 @@ def new_server(cls, server_port, client_port, tests, handler_cls=None):
             raise RuntimeError(f"TestServer failed to start due to port {server_port} already in use issue") from exc
 
     def __init__(self, server_port, client_port, tests, handler_cls=None):
+        self.request_count = 0
         self._logger = Logger(title="TestServer", print_to_console=False)
         self._client_port = client_port
         self._server_addr = ("", server_port)
-        self._server = MyHTTPServer(self._server_addr,
-                                    handler_cls or RequestHandler,
-                                    tests)
-        self._server.allow_reuse_address = True
-        self._server.timeout = TIMEOUT
-        self._thread = Thread(target=self._run)
+        self._server_error = None
+        self._is_ready = Event()
+        self._thread = Thread(target=self._run, args=(tests,
+                                                      handler_cls or RequestHandler))
         self._thread.start()
+        self._is_ready.wait(TIMEOUT)
```

Comment:
Can you please add a comment in here explaining why a `wait()` is called here and the `set()` is called from inside the thread ?